sphinx_pyreverse/__init__.py

```diff
@@ -49,16 +58,28 @@ def run(self):
         self.module_name = self.arguments[0]
         
         if self.module_name not in self.generated_modules:
-            print(call(['pyreverse', '-o', 'png', '-p', self.module_name, self.module_name], cwd=self.uml_dir))
+            cmd = ['pyreverse', '-o', 'png', '-p', self.module_name]
+            if 'show-ancestors' in self.options:
+                cmd.append("-a")
+                cmd.append(str(self.options['show-ancestors']))
+            if 'show-associated' in self.options:
+                cmd.append("-s")
+                cmd.append(str(self.options['show-associated']))
+            cmd.append(self.module_name)
+            #print("\tExecuting {}".format(" ".join(cmd)))
+            res = call(cmd, cwd=self.uml_dir)
+            #print("\tresult", res)
             # avoid double-generating
             self.generated_modules.append(self.module_name)
         
-        valid_flags = {':classes:', ':packages:'}
-        if not set(self.arguments[1:]).issubset(valid_flags):
+        # print(self.arguments)
+        valid_flags = {'classes', 'packages'}
+        flags = set(self.options) & valid_flags
+        if len(flags) != 1:
```

Comment:
This check means you can't have more than one of ':classes:' or ':packages:' being used.